File: Easy/Javascript/1207. Unique Number of Occurrences.js
/**Given an array of integers arr, return true if the number of occurrences of each value 
 * in the array is unique or false otherwise. */




/**
 * @param {number[]} arr
 * @return {boolean}
 */



var uniqueOccurrences = function(arr) {
    var mapping = {};

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] in mapping) {
            mapping[arr[i]] = mapping[arr[i]] + 1
        }
        else {
            mapping[arr[i]] = 1
        }
    }

    var visited = []

    for (key of Object.values(mapping)) {
        if (visited.includes(key)) {
            console.log('breka at', key)
            return false
        }
        visited.push(key)
    }
    return true
};

console.log(uniqueOccurrences([1,2,2,1,1,3]))

File: Easy/Python/1207. Unique Number of Occurrences.py
''' Given an array of integers arr, return true if the number of occurrences of each value in the array is unique or 
false otherwise. '''

class Solution:
    def uniqueOccurrences(self, arr: list[int]) -> bool:
        mapping = {}
        visited = []

        for i in arr:
            if i in visited:
                continue

            key = str(arr.count(i))
            visited.append(i)

            if key not in mapping:
                mapping[key] = i
            elif not mapping[key] == i:
                return False

        return True


print(Solution().uniqueOccurrences([1,2,2,1,1,3]))